nhsn/delphi_nhsn/pull.py

```diff
@@ -49,10 +48,12 @@ def check_last_updated(socrata_token, dataset_id, logger):
             )
         else:
             logger.info(f"{prelim_prefix}NHSN data is stale; Skipping", updated_timestamp=updated_timestamp)
+        return recently_updated_source
     # pylint: disable=W0703
     except Exception as e:
-        logger.info("error while processing socrata metadata; treating data as stale", error=str(e))
-    return recently_updated_source
+        logger.info("error while processing socrata metadata", error=str(e))
+        raise e
+
```

Comment:
you dont need to specify the exception to re-raise it, and the extra newline upsets the linter:
```suggestion
        logger.error("error while processing socrata metadata", error=str(e))
        raise
```